src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/WarningCaseEventHandler.java

```diff
@@ -176,6 +201,19 @@ private SendMessageRequest createOldFormatWarningMessage(String caseReference,
                 correlationKeys.put("taskCategory", categories);
             }
 
+            if (warningCode != null && warningText != null) {
```

Comment:
Suggestion: this is a duplicate block. We could reuse this code if you moved it to a method.